src/main/kotlin/com/exactpro/th2/codec/TransportCodecProcessor.kt

```diff
@@ -50,6 +53,10 @@ open class TransportCodecProcessor(
     override val MessageGroup.parsedProtocols get() = allParsedProtocols
     override val MessageGroup.eventIds get() = allParentEventIds
     override fun MessageGroup.ids(batch: GroupBatch): List<ProtoMessageID> = messageIds(batch)
+    override fun Message<*>.id(batch: GroupBatch): MessageID = id.toProto(batch)
+    override fun Message<*>.book(batch: GroupBatch): String = batch.book
+    override val Message<*>.eventBook: String? get() = eventId?.book
+    override val Message<*>.eventId: EventID? get() = eventId?.toProto()
```

Comment:
If we return MessageID and EventID in both proto and transport processors maybe we don't need `book` method and `eventBook` property? We can take this information from those objects

Reply:
I would like to avoid creating several redundant instance of protobuf objects

Reply:
Whatever you wish. I just wanted to point this out